### PR TITLE
fix wrong load reports & task leaks

### PR DESCRIPTION
**Pull request by theomonnom**
basically https://github.com/livekit/agents/pull/459

### CODE REVIEW

File: livekit-agents/livekit/agents/worker.py [hunk 1]
```diff
@@ -313,9 +314,11 @@ async def _worker_task(self) -> None:
                     break
 
                 if retry_count >= self._opts.max_retry:
-                    raise Exception(
-                        f"failed to connect to livekit after {retry_count} attempts: {e}"
+                    logger.error(
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/worker.py:317]: we don't want to throw?

Comment (theomonnom) [reply livekit-agents/livekit/agents/worker.py:317]: the reason is that the error were showing twice, because the cli also log exception on worker.run()

Comment (theomonnom) [reply livekit-agents/livekit/agents/worker.py:317]: I'll double check to see if it introduced an issue where some code was waiting for an exc

Comment (theomonnom) [reply livekit-agents/livekit/agents/worker.py:317]: yeah you're right, reverting to an exception

**Pull request merged**